src/serverless-stage-destroyer.ts

```diff
@@ -0,0 +1,324 @@
+import {
+  CloudFormationClient,
+  DescribeStacksCommand,
+  paginateDescribeStacks,
+  paginateListStackResources,
+  DeleteStackCommand,
+  waitUntilStackDeleteComplete,
+} from "@aws-sdk/client-cloudformation";
+
+import {
+  S3Client,
+  PutBucketVersioningCommand,
+  ListObjectVersionsCommand,
+  DeleteObjectsCommand,
+  HeadBucketCommand,
+  PutBucketPolicyCommand,
+} from "@aws-sdk/client-s3";
+import * as readlineSync from "readline-sync";
+
+type Tag = {
+  Key: string;
+  Value: string;
+};
+
+export class ServerlessStageDestroyer {
+  public async destroy(
+    region: string,
+    stage: string,
+    options: {
+      filters: Tag[];
+      verify: boolean;
+      wait: boolean;
+      bucketsToSkip?: string[];
+    }
```

Comment:
```suggestion
    options: {
      filters: Tag[];
      verify: boolean;
      wait: boolean;
      bucketsToSkip?: string[];
    } =  = {
    filters: [],
    verify: true,
    wait: true,
    bucketsToSkip: [],
  }
```
I think you can do this to set defaults here. But also 🤷 not sure it's better